backend/src/services/journal/journal.permissions.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+class JournalPermissions {
+  constructor(journalRepo, authRepo, coursePermissions) {
+    this.journalRepo = journalRepo;
+    this.authRepo = authRepo;
+    this.coursePermissions = coursePermissions;
+  }
+
+  /**
+   * Check if a user can access a journal entry.
+   * - If journal is private: only the owner can access
+   * - If journal is not private: owner, students in same team, and TAs managing that team can access
+   * @param {number} userId - ID of the user trying to access
+   * @param {number} journalId - ID of the journal entry
+   * @returns {Promise<boolean>} true if user can access the journal
+   */
+  async canAccessJournalEntry(userId, journalId) {
+    const journal = await this.journalRepo.getJournalById(journalId);
+    if (!journal) {
+      return false;
+    }
+
+    // Owner can always access their own journal
+    if (journal.user_id === userId) {
+      return true;
+    }
+
+    // If journal is private, only owner can access
+    if (journal.is_private === true) {
+      return false;
+    }
+
+    // Journal is not private, check if user is in same team or is a TA managing that team
+    const journalOwnerTeam =
+      await this.journalRepo.getJournalOwnerTeam(journalId);
+
+    // If journal owner is not in a team, only owner can access (already checked above)
+    if (!journalOwnerTeam) {
+      return false;
+    }
+
+    // Check if user is in the same team
+    const userTeam = await this.journalRepo.getTeamIdByUserId(
+      userId,
+      journal.course_id
+    );
+    if (userTeam === journalOwnerTeam) {
+      return true;
+    }
```

Comment:
[Question]
Do we want all team members to be able to see journals from other members of the team? Or should it only be accessible to the team lead(s)?

Reply:
Hmm.. I was thinking that this would be where the student would also leave communication specifically for professors (maybe like feedback on team's processes/ members). However, I think I am conflating the use case for journals with a course feedback system.

Let's go with your reasoning and keep public journals visible to team members.